File: server/src/main/java/ru/tinkoff/services/JobService.java
package ru.tinkoff.services;

import ru.tinkoff.exceptions.JobNotFoundException;
import ru.tinkoff.models.Job;
import ru.tinkoff.repositories.JobRepository;

public final class JobService {
    private final JobRepository jobRepository;

    public JobService() {
        this.jobRepository = new JobRepository();
    }

    public Job getJob(String code) {
        Job job = this.jobRepository.getJobByCode(code);
        if (job == null) {
            throw new JobNotFoundException(code);
        }

        return job;
    }
}

File: client/src/app/services/job.service.ts
import { Injectable } from '@angular/core';
import { Job } from '../models/job';
import { ApiClient } from './api-client';

@Injectable()
export class JobService {
    constructor(private client: ApiClient) { }

    async getJob(code: string): Promise<Job> {
        return this.client.get<Job>(`/api/jobs/${code}`);
    }
}
